gbpservice/neutron/nsf/config_agent/firewall.py

```diff
@@ -0,0 +1,154 @@
+import sys
+import ast
+
+from oslo_config import cfg
+from oslo_messaging import target
+from oslo_log import log as logging
+from neutron import manager
+
+from neutron_fwaas.db.firewall import firewall_db
+
+from gbpservice.neutron.nsf.config_agent import RestClientOverUnix as rc
+from gbpservice.neutron.nsf.config_agent import topics
+from neutron.common import rpc as n_rpc
+
+LOG = logging.getLogger(__name__)
+
+Version = 'v1'  # v1/v2/v3#
+
+
+class Firewall(object):
+    API_VERSION = '1.0'
+
+    def __init__(self):
+        self.topic = topics.FW_NSF_PLUGIN_TOPIC
+        _target = target.Target(topic=self.topic,
+                                version=self.API_VERSION)
+        n_rpc.init(cfg.CONF)
+        self.client = n_rpc.get_client(_target)
+        self.cctxt = self.client.prepare(version=self.API_VERSION,
+                                         topic=self.topic)
+
+    def set_firewall_status(self, **kwargs):
+        context = kwargs.get('context')
+        del kwargs['context']
+        self.cctxt.cast(context, 'set_firewall_status',
+                        host=kwargs['host'],
+                        firewall_id=kwargs['firewall_id'],
+                        status=kwargs['status'])
+
+    def firewall_deleted(self, **kwargs):
+        context = kwargs.get('context')
+        del kwargs['context']
+        self.cctxt.cast(context, 'firewall_deleted',
+                        host=kwargs['host'],
+                        firewall_id=kwargs['firewall_id'])
+
+
+class FwAgent(firewall_db.Firewall_db_mixin):
+
+    RPC_API_VERSION = '1.0'
+    _target = target.Target(version=RPC_API_VERSION)
+
+    def __init__(self, conf, sc):
+        self._conf = conf
+        self._sc = sc
+        super(FwAgent, self).__init__()
+
+    def _prepare_request_data(self, resource, kwargs):
+
+        request_data = {'info': {
+            'version': Version,
+            'service_type': "firewall"
+        },
+
+            'config': [{
+                'resource': resource,
+                'kwargs': kwargs
+            }]
+        }
+
+        return {'request_data': request_data}
+
+    def create_firewall(self, context, fw, host):
+
+        db = self._context(context, fw['tenant_id'])
```

Comment:
This code to create the post request and post it is common across create/delete_firewall. Make it an internal method and invoke.